docs/slide-in-use-cases.md

```diff
@@ -0,0 +1,64 @@
+# Overview through Slide In use cases
+
+It turned out that the sliding-in feature is more complex than it had appeared in the first place.
+This paper is an overview through the use cases.
+It should help to implement and understand the feature.
+
+## Slide In New Branch Dialog
+
+The picture below shows the **Slide In New Branch Dialog**.
+This is a regular **Create New Branch Dialog** with a changed title.
+
+![Slide In New Branch Dialog](slide-in-new-branch-dialog.png "Slide In New Branch Dialog")
+
+Note the checkbox options:
+- Checkout branch
+- Overwrite existing branch
+
+## Use cases
+
+| UC ID | Branch entry exists | Local branch exists | Estimated frequency |
+| :---: | :---: | :---: | :---: |
+| 1 | false | false | ~ 75% |
+| 2 | false | true | ~ 25% |
+| 3 | true | true | < 1% |
+| 4 | true | false | < 1% |
+
+## Slide In - Logic
+
+- When the **entry does not exist**, it must be added (slid in) under the given branch.
+- When the **entry exists under the given parent**, branch layout must not be affected.
+- When the **entry exists under other parent**, it must be first removed (slid out),
+and then added (slid in) under the given parent.
+
+## Slide In - New Branches
+
+This section applies to **UC1** and **UC4**.
```

Comment:
Since the cases are naturally grouped by `Local branch exists` (so UC1&UC4 vs UC2&UC3) rather than by `Branch entry exists`, maybe reorder them (as, in swap the columns of the table)